resteasy reactive doesn't support wildcard type since quarkus 2.7.2

### Describe the bug

Hi,

I have this method : 

```java
@GET
    @Produces(MediaType.TEXT_PLAIN)
    public Uni<RestResponse<?>> decodeKVB() {
        return connector.connectAsync() ;
    }
```

It works well until quarkus 2.7.1
Since I upgrade to quarkus 2.7.2 / 2.7.3 or 2.7.4 my application fails to start with the following stacktrace : 

```java
2022-03-10 18:33:03,434 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        ... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 15 more
Caused by: java.lang.UnsupportedOperationException: Endpoint return type not supported yet: org.jboss.resteasy.reactive.common.util.types.WildcardTypeImpl@6c5e6ae4
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.getEffectiveReturnType(RuntimeResourceDeployment.java:647)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.getEffectiveReturnType(RuntimeResourceDeployment.java:643)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.getEffectiveReturnType(RuntimeResourceDeployment.java:637)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.buildResourceMethod(RuntimeResourceDeployment.java:361)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeDeploymentManager.deploy(RuntimeDeploymentManager.java:112)
        at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder.createDeployment(ResteasyReactiveRecorder.java:121)
        at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment1557341086.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment1557341086.deploy(Unknown Source)
        ... 16 more


2022-03-10 18:33:04,243 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: java.lang.UnsupportedOperationException: Endpoint return type not supported yet: org.jboss.resteasy.reactive.common.util.types.WildcardTypeImpl@6c5e6ae4
        at io.quarkus.dev.appstate.ApplicationStateNotification.waitForApplicationStart(ApplicationStateNotification.java:51)
        at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:122)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:144)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:455)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:66)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:140)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:96)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:132)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.UnsupportedOperationException: Endpoint return type not supported yet: org.jboss.resteasy.reactive.common.util.types.WildcardTypeImpl@6c5e6ae4
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.getEffectiveReturnType(RuntimeResourceDeployment.java:647)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.getEffectiveReturnType(RuntimeResourceDeployment.java:643)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.getEffectiveReturnType(RuntimeResourceDeployment.java:637)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.buildResourceMethod(RuntimeResourceDeployment.java:361)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeDeploymentManager.deploy(RuntimeDeploymentManager.java:112)
        at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder.createDeployment(ResteasyReactiveRecorder.java:121)
        at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment1557341086.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment1557341086.deploy(Unknown Source)
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

Thank you for help

Bertrand



### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

1 : create an app with the quarkus-resteasy-reactive-jackson dependency
2 : create a REST WS with a method that returns a Uni<RestResponse<?>>
3 : try to start the application

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "11" 2018-09-25 OpenJDK Runtime Environment 18.9 (build 11+28) OpenJDK 64-Bit Server VM 18.9 (build 11+28, mixed mode)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d) Maven home: C:\dev\apache-maven-3.6.3\bin\.. Java version: 11, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-11 Default locale: fr_FR, platform encoding: Cp1252 OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeResourceDeployment.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeResourceDeployment.java
@@ -8,6 +8,7 @@ import java.lang.annotation.Annotation;
 import java.lang.reflect.Method;
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
+import java.lang.reflect.WildcardType;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Arrays;
@@ -631,19 +632,26 @@ public class RuntimeResourceDeployment {
             return returnType;
         if (returnType instanceof ParameterizedType) {
             ParameterizedType type = (ParameterizedType) returnType;
+            Type firstTypeArgument = type.getActualTypeArguments()[0];
             if (type.getRawType() == CompletionStage.class) {
-                return getEffectiveReturnType(type.getActualTypeArguments()[0]);
+                return getEffectiveReturnType(firstTypeArgument);
             }
             if (type.getRawType() == Uni.class) {
-                return getEffectiveReturnType(type.getActualTypeArguments()[0]);
+                return getEffectiveReturnType(firstTypeArgument);
             }
             if (type.getRawType() == Multi.class) {
-                return getEffectiveReturnType(type.getActualTypeArguments()[0]);
+                return getEffectiveReturnType(firstTypeArgument);
             }
             if (type.getRawType() == RestResponse.class) {
-                return getEffectiveReturnType(type.getActualTypeArguments()[0]);
+                return getEffectiveReturnType(firstTypeArgument);
             }
             return returnType;
+        }
+        if (returnType instanceof WildcardType) {
+            Type[] bounds = ((WildcardType) returnType).getLowerBounds();
+            if (bounds.length > 0)
+                return getRawType(bounds[0]);
+            return getRawType(((WildcardType) returnType).getUpperBounds()[0]);
         }
         throw new UnsupportedOperationException("Endpoint return type not supported yet: " + returnType);
     }

--- a/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/response/RestResponseResource.java
+++ b/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/response/RestResponseResource.java
@@ -27,6 +27,12 @@ public class RestResponseResource {
     }
 
     @GET
+    @Path("rest-response-wildcard")
+    public RestResponse<?> wildcard() {
+        return RestResponse.ResponseBuilder.ok("Hello").header("content-type", "text/plain").build();
+    }
+
+    @GET
     @Path("rest-response-full")
     public RestResponse<String> getResponse() throws URISyntaxException {
         CacheControl cc = new CacheControl();

--- a/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/response/RestResponseTest.java
+++ b/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/response/RestResponseTest.java
@@ -27,6 +27,10 @@ public class RestResponseTest {
                 .then().statusCode(200)
                 .and().body(Matchers.equalTo("Hello"))
                 .and().contentType("text/plain");
+        RestAssured.get("/rest-response-wildcard")
+                .then().statusCode(200)
+                .and().body(Matchers.equalTo("Hello"))
+                .and().contentType("text/plain");
         RestAssured.get("/rest-response-full")
                 .then()
                 .statusCode(200)

